File: test-java/src/test/java/com/example/MainTest.java
package com.example;

import com.apollographql.java.client.ApolloClient;
import mockwebserver3.MockResponse;
import mockwebserver3.MockWebServer;
import okhttp3.Headers;
import org.junit.Test;

import java.io.IOException;

public class MainTest {
    @Test
    public void test() throws IOException {

        try(MockWebServer mockServer = new MockWebServer()) {
            mockServer.start();
            mockServer.enqueue(new MockResponse(200, Headers.EMPTY, "{\"data\": {\"foo\": \"bar\"}}"));

            try (ApolloClient apolloClient = new ApolloClient.Builder()
                    .serverUrl(mockServer.url("/").toString())
                    .build()) {

                apolloClient.query(new GetFooQuery()).enqueue(System.out::println);
            }
        }
    }
}

File: apollo-6789/src/test/kotlin/MainTest.kt
import com.apollographql.apollo.ApolloClient
import foo.ExampleQuery
import kotlinx.coroutines.runBlocking
import kotlin.test.Test

class MainTest {
  @Test
  fun test() {
    runBlocking {
      val apollo = ApolloClient.Builder().build()
      apollo.query(ExampleQuery())
        .execute()
        .dataOrThrow()
        .foo
        .bar
        .id // bar is @catch(to: THROW) and bar is non-null
    }
  }
}